src/tests/Register.test.js

```diff
@@ -1,11 +1,79 @@
 import React from 'react';
-import ShallowRenderer from 'react-test-renderer/shallow';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { render, fireEvent, screen, waitForElement, waitFor } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import Register from "../register/Register";
 
 
-test('renders Register', () => {
-    const renderer = new ShallowRenderer();
-    renderer.render(<Register />);
-    expect(renderer.getRenderOutput()).toMatchSnapshot()
-});
\ No newline at end of file
+// Mock successful registration
+const server = setupServer(
+    rest.post('http://127.0.0.1:5000/register', (req, res, ctx) => {
+        return res(ctx.json({ message: "User was created successfully. A confirmation email has been sent via email. After confirming your email you can login." }))
+    })
+)
+
+beforeAll(() => server.listen())
+afterEach(() => server.resetHandlers())
+afterAll(() => server.close())
+
+it('allows the user to register successfully', async () => {
+    render(<Register />)
+
+    fireEvent.change(screen.getByPlaceholderText('Full Name'), {
+        target: { value: 'My name' },
+    })
+    fireEvent.change(screen.getByPlaceholderText('Username'), {
+        target: { value: 'MyUsername' },
+    })
+    fireEvent.change(screen.getByPlaceholderText('Email'), {
+        target: { value: 'myemail@xxx.xxx' },
+    })
+    fireEvent.change(screen.getByPlaceholderText('Password'), {
+        target: { value: '12345678' },
+    })
+    fireEvent.change(screen.getByRole('checkbox', { name: "mentor" }), {
+        target: { value: 'true' },
+    })
```

Comment:
fireEvent.change doesn't work on check boxes. You can cross check this by doing the following:
`expect(screen.getByRole('checkbox', {name: "mentor"}).checked).toBe(true)`

and it will fail. So click works on checkboxes, check this out - https://stackoverflow.com/questions/55177928/how-do-you-check-a-checkbox-in-react-testing-library

Reply:
done. changed it